pkgs/applications/networking/cluster/hadoop/default.nix

```diff
@@ -17,6 +17,8 @@
 , openssl
 , glibc
 , nixosTests
+, sparkSupport ? true
```

Comment:
Also about the other flags: If they are true by default. Is there a usecase where you want to disable the feature and is that likely to be used? If it is just a nice to have we don't really need them and less complexity is always great, especially if it is not well tested.

Reply:
Most changes made by me around hadoop/spark are based on changes made to an internal fork of nixpkgs for production infrastructure. Features aren't added to modules and packages without a good use-case. In this case, for example, Hadoop is a standalone package that doesn't need spark. In most cases, spark support is useful to have, hence its inclusion by default. There are two situations where you want to remove spark to either limit a YARN cluster to running only mapreduce jobs, or simply to deploy a more lightweight hadoop package where spark is not needed.